src/Darryldecode/Cart/Cart.php

```diff
@@ -784,4 +755,14 @@ public function setThousandsSep($thousands_sep)
     {
         $this->thousands_sep = $thousands_sep;
     }
+
+    /**
+     * @param $name
+     * @param $value
+     * @return mixed
+     */
+    protected function fireEvent($name, $value = [])
+    {
+        return $this->events->fire($this->getInstanceName() . '.' . $name, array_values([$value, $this]), true);
```

Comment:
Event halting shouldn't be true otherwise only the first listener will be executed.

See _Illuminate\Events\Dispatcher_ 
```
public function fire($event, $payload = [], $halt = false)
{

    /*** Omitted code ***/

    // If a response is returned from the listener and event halting is enabled
    // we will just return this response, and not call the rest of the event
    // listeners. Otherwise we will add the response on the response list.
    if (! is_null($response) && $halt) {
        array_pop($this->firing);
        return $response;
    }

    /*** Omitted code ***/

}
```